src/Test/Perf/Utilities/RelativeDirectory.cs

```diff
@@ -9,44 +9,66 @@ namespace Roslyn.Test.Performance.Utilities
 {
     public class RelativeDirectory
     {
-        string _workingDir;
+        private string _workingDir;
 
-        public RelativeDirectory([CallerFilePath] string workingFile = "")
+
+        public RelativeDirectory()
         {
-            _workingDir = Directory.GetParent(workingFile).FullName;
         }
 
-        public string MyWorkingDirectory => _workingDir;
+        public RelativeDirectory(string workingDir)
+        {
+            _workingDir = workingDir;
+        }
 
-        public string TempDirectory
+        public void SetWorkingDirectory(string workingDir)
+        {
+            _workingDir = workingDir;
+        }
+
+        private void ThrowIfNotSetup()
+        {
+            if (_workingDir == null)
```

Comment:
So that it can be set from outside the object via SetWorkingDirectory.  This is done in Perf.Runner to prepare the tests for execution.